src/renderer/components/tree-view/tree-view.tsx

```diff
@@ -0,0 +1,102 @@
+import "./tree-view.scss";
+
+import React, { useEffect, useRef } from "react";
+import { Icon } from "../icon";
+import TreeView from "@material-ui/lab/TreeView";
+import TreeItem from "@material-ui/lab/TreeItem";
+import { cssNames } from "../../utils";
+
+import _ from "lodash";
+import getDeepDash from "deepdash";
+
+const deepDash = getDeepDash(_);
+
+export interface NavigationTree {
+  id: string;
+  parentId: string;
+  name: string;
+  selected?: boolean;
+  children?: NavigationTree[];
+}
+
+interface Props {
+  data: NavigationTree[]
+}
+
+const scrollToItem = (id: string) => {
+  const element = document.getElementById(id);
+
+  element?.scrollIntoView();
+};
+
+const getSelectedNode = (data: NavigationTree[]) => {
+  return deepDash.findDeep(data, (value, key) => key === "selected" && value === true)?.parent;
+};
+
+export function RecursiveTreeView({ data }: Props) {
+  const [expanded, setExpanded] = React.useState<string[]>([]);
+  const prevData = useRef<NavigationTree[]>(data);
+
+  const handleToggle = (event: React.ChangeEvent<{}>, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  const expandTopLevelNodes = () => {
+    setExpanded(data.map(node => node.id));
+  };
+
+  const expandParentNode = () => {
+    const node = getSelectedNode(data) as any as NavigationTree;
+    const id = node?.parentId;
```

Comment:
The above allows this to be:
```suggestion
    const id = getSelectedNode(data)?.parentId;
```

Reply:
Reverted `getSelectedNode`  into a function but left types untouched since `.parentId` also needs same kind of `as any...` transformations.
```
Property 'parentId' does not exist on type 'unknown'.ts(2339)
```

Reply:
I would say that `getSelectedNode`'s return value being `IDeepEntry` (though if you look at the typings even that is wrong) is an implementation leak. The cast should be within `getSelectedNode`.

Reply:
Yes, unfortunately, deepDash has broken typings. I was using [type assertions[(https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) here and unable to find an option to set proper types for return value of `getSelectedNode`

Reply:
I meant having the `as any as NavigationTree` within your helper function but that is of low priority.